docs/optlib.rst

```diff
@@ -181,269 +206,413 @@ Tips about writing option file
-	it is important to balance `push` and `pop`.
+	Don't print a tag captured with a regex pattern to a tag file. This is
+	useful when you need to push non-named context information to the stack.
+	Well known non-named scope in C language is established with `{`. A non-
+	named scope never appears in tags file as a name or scope name.  However,
+	pushing it is important to balance ``push`` and ``pop``.
 
-Example 1::
+Example 1:
 
-	$ cat /tmp/input.foo
+.. code-block:: python
+
+	# in /tmp/input.foo
 	class foo:
 	def bar(baz):
 		print(baz)
 	class goo:
 	def gar(gaz):
 		print(gaz)
 
-	$ cat /tmp/foo.ctags
-	--langdef=foo
-		--map-foo=+.foo
-		--regex-foo=/^class[[:blank:]]+([[:alpha:]]+):/\1/c,class/{scope=set}
-		--regex-foo=/^[[:blank:]]+def[[:blank:]]+([[:alpha:]]+).*:/\1/d,definition/{scope=ref}
+.. code-block:: perl
+
+	# in /tmp/foo.ctags:
+	--langdef=Foo
+	--map-Foo=+.foo
+	
+	--regex-Foo=/^class[[:blank:]]+([[:alpha:]]+):/\1/c,class/{scope=set}
+	--regex-Foo=/^[[:blank:]]+def[[:blank:]]+([[:alpha:]]+).*:/\1/d,definition/{scope=ref}
 
-	$ ~/var/ctags/ctags --options=/tmp/foo.ctags -o - /tmp/input.foo
+.. code-block:: console
+
+	$ ctags --options=/tmp/foo.ctags -o - /tmp/input.foo
 	bar	/tmp/input.foo	/^    def bar(baz):$/;"	d	class:foo
 	foo	/tmp/input.foo	/^class foo:$/;"	c
 	gar	/tmp/input.foo	/^    def gar(gaz):$/;"	d	class:goo
 	goo	/tmp/input.foo	/^class goo:$/;"	c
 
 
-Example 2::
+Example 2:
+
+.. code-block:: c
 
-	$ cat /tmp/input.pp
+	// in /tmp/input.pp
 	class foo {
-	include bar
+		int bar;
 	}
 
-	$ cat /tmp/pp.ctags
+.. code-block:: perl
+
+	# in /tmp/pp.ctags:
 	--langdef=pp
-		--map-pp=+.pp
-		--regex-pp=/^class[[:blank:]]*([[:alnum:]]+)[[[:blank:]]]*\{/\1/c,class,classes/{scope=push}
-		--regex-pp=/^[[:blank:]]*include[[:blank:]]*([[:alnum:]]+).*/\1/i,include,includes/{scope=ref}
-		--regex-pp=/^[[:blank:]]*\}.*//{scope=pop}{exclusive}
+	--map-pp=+.pp
 
-	$ ~/var/ctags/ctags --options=/tmp/pp.ctags -o - /tmp/input.pp
-	bar	/tmp/input.pp	/^    include bar$/;"	i	class:foo
-	foo	/tmp/input.pp	/^class foo {$/;"	c
+	--regex-pp=/^[[:blank:]]*\}//{scope=pop}{exclusive}
+	--regex-pp=/^class[[:blank:]]*([[:alnum:]]+)[[[:blank:]]]*\{/\1/c,class,classes/{scope=push}
+	--regex-pp=/^[[:blank:]]*int[[:blank:]]*([[:alnum:]]+)/\1/v,variable,variables/{scope=ref}
 
+.. code-block:: console
+
+	$ ctags --options=/tmp/pp.ctags -o - /tmp/input.pp
+	bar	/tmp/input.pp	/^    include bar$/;"	v	class:foo
+	foo	/tmp/input.pp	/^class foo {$/;"	c
 
-NOTE: Giving a scope long flag implies setting `useCork` of the parser
-to `TRUE`. See `cork API`.
 
 NOTE: This flag doesn't work well with ``--mline-regex-<LANG>=``.
 
 
-Override the letter for file kind
+Overriding the letter for file kind
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-(See also #317.)
 
+.. Q: this was fixed in https://github.com/universal-ctags/ctags/pull/331
+	so can we remove this section?
+
```

Comment:
Yes, we can remove this section because I wrote this to man/ctags-incompatibilities.7.rst.in.
I marked `.. IN MAN PAGE` when I wrote the topic to our man pages.
In the case, we can remove it. I planed removing all the marked topic when releasing 6.0.
However, you can remove some of them now.